File: Daily Challenge/Java/Matrix - Integers And Characters Column 23.6.2025.java
import java.util.*;

public class Main {
    public static boolean isDigitCol(String[] col) {
        for (String val : col) {
            for (char ch : val.toCharArray())
                if (!Character.isDigit(ch)) return false;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int r = sc.nextInt(), c = sc.nextInt();
        String[][] mat = new String[r][c];
        for (int i = 0; i < r; i++)
            for (int j = 0; j < c; j++)
                mat[i][j] = sc.next();

        List<String> nums = new ArrayList<>();
        List<String> chars = new ArrayList<>();

        for (int j = 0; j < c; j++) {
            String[] col = new String[r];
            for (int i = 0; i < r; i++)
                col[i] = mat[i][j];

            if (isDigitCol(col)) {
                int sum = 0;
                for (String val : col) sum += Integer.parseInt(val);
                nums.add(String.valueOf(sum));
            } else {
                StringBuilder sb = new StringBuilder();
                for (int i = r - 1; i >= 0; i--)
                    sb.append(mat[i][j]);
                chars.add(sb.toString());
            }
        }

        for (String s : nums) System.out.print(s + " ");
        for (String s : chars) System.out.print(s + " ");
    }
}

File: Daily Challenge/C/Matrix - Integers And Characters Column 23.6.2025.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int is_digit_col(char arr[][50], int r) {
    for (int i = 0; i < r; i++) {
        for (int j = 0; arr[i][j]; j++) {
            if (!isdigit(arr[i][j])) return 0;
        }
    }
    return 1;
}

int main() {
    int r, c;
    scanf("%d %d", &r, &c);
    char mat[50][50][50];
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            scanf("%s", mat[i][j]);

    char result[1000] = "";
    for (int j = 0; j < c; j++) {
        char col[50][50];
        for (int i = 0; i < r; i++)
            strcpy(col[i], mat[i][j]);

        if (is_digit_col(col, r)) {
            int sum = 0;
            for (int i = 0; i < r; i++)
                sum += atoi(col[i]);
            char tmp[20];
            sprintf(tmp, "%d ", sum);
            strcat(result, tmp);
        }
    }

    for (int j = 0; j < c; j++) {
        char col[50][50];
        for (int i = 0; i < r; i++)
            strcpy(col[i], mat[i][j]);

        if (!is_digit_col(col, r)) {
            for (int i = r - 1; i >= 0; i--)
                strcat(result, col[i]);
            strcat(result, " ");
        }
    }

    printf("%s\n", result);
    return 0;
}